public/action/ad_dashboard_action.tsx

```diff
@@ -0,0 +1,74 @@
+import { IEmbeddable } from '../../../../src/plugins/dashboard/public/embeddable_plugin';
+import {
+  DASHBOARD_CONTAINER_TYPE,
+  DashboardContainer,
+} from '../../../../src/plugins/dashboard/public';
+import {
+  IncompatibleActionError,
+  createAction,
+  Action,
+} from '../../../../src/plugins/ui_actions/public';
+import { isReferenceOrValueEmbeddable } from '../../../../src/plugins/embeddable/public';
+import { EuiIconType } from '@elastic/eui/src/components/icon/icon';
+
+export const ACTION_AD = 'ad';
+
+function isDashboard(
+  embeddable: IEmbeddable
+): embeddable is DashboardContainer {
+  return embeddable.type === DASHBOARD_CONTAINER_TYPE;
+}
+
+export interface ActionContext {
+  embeddable: IEmbeddable;
+}
+
+export interface CreateOptions {
+  grouping: Action['grouping'];
+  title: string;
+  icon: EuiIconType;
+  id: string;
+  order: number;
+  onClick: Function;
+}
```

Comment:
this seems kind of weird, since its mostly just fields that are in `Action`, but then `onClick` is something extra. Suggest to just have a class file that defines the action and implements `Action` interface. This way multiple fns aren't needed, and if `Action` has any changes, we can quickly figure out the error and what would need to be done to have our defined action working again.

Reply:
For example: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/embeddable/public/lib/actions/edit_panel_action.ts#L60

Reply:
Since we are doing something similar on the alerting side, I am responding with that in mind.

This solution seems similar to having a class file, but maybe even simpler, so I'm not sure how that would save us from multiple fns. If there are any changes to `createAction`, which comes from UI actions, I think that would also be an indicator of issues in our code (similar to the benefit of using an Action interface). It seems like both approaches do similar things, but here we are wrapping around `createAction` to save us some code repetition.

I think I would need to see a fuller example to understand the benefits, because looking at that example above, this solution is easier to follow. From my understanding `createAction` is a good fit for these situations.